### ISSUE TITLE
Problème affichage carto lors ajout d'adresse Geoportail

### ISSUE THREAD

**Issue by SDIS83Loic** [SDIS62/prevarisc#693]
Bonjour,

depuis la mise à jour du Géoportail nous rencontrons un problème d'affichage cartographique lors de l'ajout d'une localisation "geolocaliser cette adresse geoportail".

Dans la console du navigateur, j'ai identifié cette erreur.
https://api.ign.fr/geoportail/api/js/2.1.2/GeoportalExtended.js

`Uncaught TypeError: Cannot read property 'replace' of undefined
    at Function.Geoportal.Catalogue.completeConfiguration (GeoportalExtended.js:138)
    at Geoportal.GeoRMHandler.parseAutoConf (GeoportalExtended.js:138)
    at GeoportalExtended.js:138
    at initialize.handleResponse (GeoportalExtended.js:138)
    at initialize.handleRead (GeoportalExtended.js:138)
    at initialize.<anonymous> (GeoportalExtended.js:138)
    at Object.<anonymous> (GeoportalExtended.js:138)
    at Object.b.registry.(anonymous function) (https://api.ign.fr/geoportail/api/js/2.1.2/GeoportalExtended.js:138:1475728)
    at wxs.ign.fr/xxxxxxxxxxxxxxxxxx/autoconf/?output=json&callback=OpenLayers.Protocol.Script.registry.regId1:1`

Une idée ?

Loic

**Comment by pcano65** [comment]
Bonjour,

Nous nous n'avons plus d'affichage carto même dans la consultation des ERP.
Merci.

**Comment by SDIS91** [comment]
Bonjour

L'API v2 chez IGN est maintenant obsolète. Nous avons migré vers le SDK pour résoudre ce problème:
https://geoservices.ign.fr/documentation/utilisation_web/migration.html.
Pour plus d'information: https://geoservices.ign.fr/blog/

J'éspère que cette info sera utile.

**Comment by SDIS83Loic** [comment]
Bonjour,

Est-ce que quelqu'un a déjà fait cette migration et aurait fait un dépôt Git ? @aechinard

Merci

**Comment by aechinard** [comment]
Bonjour,

J'ai réalisé cette migration sur la version SDK version 2.0.1. Normalement, mon dépôt Git est à jour pour l'utilisation de ce SDK. Référez-vous à ce post pour plus d'infos : 
[https://github.com/SDIS62/prevarisc/issues/631]

**Comment by SDIS83Loic** [comment]
Bonjour @aechinard,

Justement je relisais ce post afin de comprendre quel fichier nous avions oublié.
Nous avions déjà fait la modification du JDK à la suite du [https://github.com//issues/631]
Mais lors de l'ajout d'une localisation la carto ne s'affiche plus. Je pense que je n'ai pas mis à jour le fichier edit.phtml.

D'autre part, quelles différences il y a-t-il entre les 2 JS ?
/js/sdk-ol/GpOL3Itowns.js
/js/sdk-ol/GpSDK3D.js
Hormis la gestion de la 3D, est-il nécessaire ou pouvons-nous rester sur GpOL3Itowns.js.

Merci
Loic

**Comment by aechinard** [comment]
Bonjour,

Effectivement il n'est pas nécessaire de conserver la gestion de la 3D mais dans une perspective d'évolution, l'option est gérée. Pour la carte affichée lors de l'ajout de l'adresse d'un établissement, je n'ai réalisé aucune modification à l'heure actuelle. Je vais appliquer les mêmes modifications que sur la gestion des couches cartographiques et réaliser le travail rapidement.

**Comment by devottered** [comment]
Bonjour à tous,

J'ai fait la migration du SDK comme Loic grâce [au travail d'aechinard](https://github.com/SDIS62/prevarisc/issues/631), et je rencontre également un problème d'affichage de la carto lors de l'ajout d'une localisation quand je crée un établissement.

@SDIS83Loic, avez-vous résolu votre problème en mettant à jour le fichier edit.phtml ?
Sur le dépôt Git de @aechinard, ce fichier n'utilise pas le SDK. (Ou alors ai-je raté quelque chose ?)

Merci d'avance pour vos réponses,
Maxime

**Comment by aechinard** [comment]
Effectivement, je n'ai rien touché à la carto pour l'ajout d'un établissement. Je regarde cela rapidement.

**Comment by devottered** [comment]
Merci pour cette réponse rapide !

Je suis également en train de regarder cela. Je reviens vers vous dès que j'ai quelque chose.

**Comment by SDIS83Loic** [comment]
Bonjour à tous,
j'ai contacté M. VANCAMP du SDIS62 pour signaler ce problème majeur qui doit toucher la communauté des utilisateurs Prevarisc.
Il a été informé du travail déjà effectué par @aechinard.
Il serait intéressant pour tous de pouvoir intégrer à la source officiel le travail de @aechinard sur la partie IGN.

**Comment by devottered** [comment]
Petit update,

J'arrive à afficher la carto lors de la saisie d'une localisation. Quelques petits soucis persistent une fois la saisie effectuée.
Je vous tiens informés.

Maxime

**Comment by devottered** [comment]
Bonsoir à tous,

Il semblerait qu'il y ait un problème au niveau des projections lors de l'ajout de localisation quand on crée un établissement.
La carto est fonctionnelle dans la modal, mais n'affiche pas la bonne localisation quand on veut visionner l'établissement en question. Ce point est en train d'être traité.

J'ai mis à jour le fichier edit.phtml pour ceux que ça pourrait intéresser quand même, [juste ici](https://github.com/MaximeMerrienAtos/prevarisc/tree/2.x/application/views/scripts/etablissement) !

Vous souhaitant une bonne soirée,
Maxime

**Comment by aechinard** [comment]
Bonjour à tous,

J’étais partis sur la même base de modification avec un petit souci d’affichage de la carte.
Le problème de décalage vient de la projection utilisée, je regarde cela et vous tiens au courant.

**Comment by devottered** [comment]
Bonjour à tous,

Je viens de corriger le problème de projection au niveau de la carto lors d'ajout d'établissement.
La visualisation après ajout est fonctionnelle de mon côté. Le fichier à jour est [ici](https://github.com/MaximeMerrienAtos/prevarisc/tree/2.x/application/views/scripts/etablissement) !

Si quelqu'un voit un problème, relancez la discussion.
Bonne journée,

Cordialement,
Maxime

**Comment by pcano65** [comment]
Bonjour,

Merci pour votre travail à tous.
J'ai fait les mises à jour et j'ai bien la carto qui est revenu sur la fiche établisssement.
Cependant, je n'ai toujours pas la carto lors je veux ajouter une adresse et je n'arrive pas à rajouter un flux wms type ortho de géoportail.
Par ailleurs si je clique sur lister les couches IGN, rien ne s'affiche.

Est ce normal ou est ce que j'ai loupé quelque chose ?

Merci!
